roles/hosted_cluster/tasks/main.yml

```diff
@@ -78,26 +78,82 @@
         infrastructureAvailabilityPolicy: HighlyAvailable
         olmCatalogPlacement: management
 
-- name: Manage NodePool resource
+- name: Manage NodePool resources
   kubernetes.core.k8s:
     state: "{{ hosted_cluster_state }}"
     definition:
       apiVersion: hypershift.openshift.io/v1beta1
       kind: NodePool
       metadata:
-        name: "nodepool-{{ hosted_cluster_name }}"
+        name: "nodepool-{{ hosted_cluster_name }}-{{ item.resource_class }}"
         namespace: "{{ hosted_cluster_namespace }}"
         labels: "{{ hosted_cluster_default_cluster_order_label }}"
       spec:
         clusterName: "{{ hosted_cluster_name }}"
-        replicas: "{{ hosted_cluster_node_count | int }}"
+        replicas: "{{ item.number_of_nodes | int }}"
         management:
           autoRepair: false
           upgradeType: InPlace
         platform:
           type: Agent
           agent:
             agentLabelSelector:
-              matchLabels: "{{ hosted_cluster_default_cluster_order_label }}"
+              matchLabels: >
+                "{{ {esi_agent_resource_class_label: item.resource_class} | combine(hosted_cluster_default_cluster_order_label) }}"
         release:
           image: "{{ ocp_release_image }}"
+  loop: "{{ hosted_cluster_node_requests }}"
+  loop_control:
+    label: "Manage NodePool nodepool-{{ hosted_cluster_name }}-{{ item.resource_class }}"
+
+- name: Create cluster network
+  ansible.builtin.include_role:
+    name: massopencloud.esi.network
+  vars:
+    network_suffix: "{{ hosted_cluster_name }}"
+    network_state: "{{ hosted_cluster_state }}"
+  when: hosted_cluster_state == "present"
+
+- name: Select and label new Agents resources
+  ansible.builtin.include_role:
+    name: manage_agents
+    tasks_from: select_and_label_new_agents
+  vars:
+    manage_agents_cluster_order_name: "{{ hosted_cluster_name }}"
+    manage_agents_desired_count: "{{ item.number_of_nodes | int }}"
+    manage_agents_resource_class: "{{ item.resource_class }}"
+  loop: "{{ hosted_cluster_node_requests }}"
+
+- name: Attach agents to cluster network and approve them
+  ansible.builtin.include_role:
+    name: manage_agents
+    tasks_from: attach_and_approve_all_new_agents
+  vars:
+    manage_agents_cluster_order_name: "{{ hosted_cluster_name }}"
+    manage_agents_cluster_network: "{{ create_network_result }}"
+
+- name: Wait for the Agents to be removed from the cluster
+  ansible.builtin.include_role:
+    name: manage_agents
+    tasks_from: wait_for_agents_to_be_removed
+  vars:
+    manage_agents_cluster_order_name: "{{ hosted_cluster_name }}"
+    manage_agents_desired_count: "{{ item.number_of_nodes | int }}"
+    manage_agents_resource_class: "{{ item.resource_class }}"
+  loop: "{{ hosted_cluster_node_requests }}"
```

Comment:
I am a little confused about what is happening here...if we are creating a clsuter, why are we waiting for agents to be removed?

Reply:
the `hosted_cluster` manages creation and deletion of clusters based on `hosted_cluster_state`